src/coreComponents/mesh/Perforation.cpp

```diff
@@ -59,8 +73,59 @@ void Perforation::postInputInitialization()
   GEOS_ERROR_IF( m_distanceFromHead <= 0,
                  getWrapperDataContext( viewKeyStruct::distanceFromHeadString() ) <<
                  ": distance from well head to perforation cannot be negative." );
-}
 
+  // Setup perforation status function
+  FunctionManager & functionManager = FunctionManager::getInstance();
+
+  if( m_perfStatusTableName.empty() )
+  {
+    // No table function provide, defaults name to perforation name
+    m_perfStatusTableName=getName();
+  }
+  else
+  {
+    // Table name provide as input , check that it exists
+    GEOS_THROW_IF( !functionManager.hasGroup< TableFunction >( m_perfStatusTableName ),
+                   GEOS_FMT( "Perforation status table function missing  : {}", m_perfStatusTableName ),
```

Comment:
To add the XML file line in the error:
```suggestion
                   GEOS_FMT( "{}: Perforation status table function missing  : {}",
                             getDataContext(), m_perfStatusTableName ),
```